### ISSUE TITLE
Make the "receive" function available to caller

### ISSUE THREAD

**Issue by Moxide** [mike-north/ember-phoenix#46]
Hello,
I just started playing with phoenix's channels and found a small issue with this library.
I try to get information from the reply of the "join" call. According to the doc, this could be done using the ```receive``` function :
```
channel.join()
  .receive("ok", resp => { console.log("Joined successfully", resp) })
```
But when I try to use ```ember-phoenix``` :
```
const channel = this.joinChannel("room:123", {
  nickname: "Mike"
  })
  .receive("ok", resp => { console.log("Joined successfully", resp) });
```
this throws a "receive function is not defined"
So I had to make ```ember-phoenix``` return the ```channel.join()``` object instead of the ```channel``` object, and it works. But now, to access the ```channel``` object itself, I have to do this :
```
    const channel = this.joinChannel("room:123", {
      nickname: "Mike"
    });
    // add message handlers
    channel.channel.on("notification", () => _onNotification(...arguments));
```

Is there a more "elegant" way to address this issue ?
Thanks !

**Comment by juanazam** [comment]
@Moxide I'm facing the same issue, I guess this could be handled in two ways, we can either modify `joinChannel` to return something like `{ channel: channel, joinObject: channel.join}` or we can include a handler function on the `joinChannel` function definition and add an "ok" handler within the method.

```js
  joinChannel(name, params, okHandler) {
    ...
    channel
      .join()
      .receive("ok", okHandler);
    return channel;
  }
```

Let's see what's @mike-north take on this.

**Comment by Moxide** [comment]
Hi,
Actually the channel object is already present in the object returned by channel.join() ;-)
The 'okHandler' method looks prettier, thanks !

**Comment by juanazam** [comment]
Yes, I know the returned object contains the channel, but if we change the return value of the `joinChannel` function, users will have to modify their code, the api would change.

**Comment by Moxide** [comment]
ok, got it ;-)

**Comment by greyhwndz** [comment]
any say on this? @mike-north

**Comment by bkbooth** [comment]
I just ran into this issue. I ended up writing my own `joinChannel()` method in my service that extends `PhoenixSocket` and made it return a `Promise` so that I can know if the connection worked or not:

```
joinChannel(name, params) {
  return new Ember.RSVP.Promise((resolve, reject) => {
    const socket = get(this, 'socket')
    Ember.assert('Must connect to a socket first', socket)

    const channel = socket.channel(name, params)
    channel.join()
      .receive('ok', () => resolve(channel))
      .receive('error', (message) => reject(new Error(message)))
  })
},
```
